library/src/main/java/com/novoda/downloadmanager/RoomAppDatabase.java

```diff
@@ -1,12 +1,31 @@
 package com.novoda.downloadmanager;
 
 import android.arch.persistence.room.Database;
+import android.arch.persistence.room.Room;
 import android.arch.persistence.room.RoomDatabase;
+import android.content.Context;
 
 @Database(entities = {RoomBatch.class, RoomFile.class}, version = 1)
 abstract class RoomAppDatabase extends RoomDatabase {
 
+    private static RoomAppDatabase SINGLE_INSTANCE;
```

Comment:
It's possible that we can be using the `RoomAppDatabase` instance in two places at once now - once in the migrator and once in the actual downloads service, if the user decides to initiate a download while the migration process is happening. We will need to test this scenario, but for now, we followed the Room guidelines and made this a singleton.